### PR TITLE
Make replace and replace_first deep

### PR DESCRIPTION
**Pull request by ysthakur**
Trying to implement #1398. I'm not entirely sure what @Steffan153 wanted, but I made `replace` deep, hopefully that's fine.

### CODE REVIEW

**Comment by Steffan153** [comment]
Rip this is not at all what I wanted lmao.

**Comment by Steffan153** [comment]
so like `"abcdef", ["b", "d"], "z"` => `azczef`, and `"abcdef", ["b", "d"], ["x", "y"]` => `axcyef`

File: vyxal/helpers.py [hunk 1]
```diff
@@ -927,6 +927,16 @@ def primitive_type(item: Any) -> Union[str, type]:
     return list
 
 
+def replace_deep(
+    lst: VyList, target: Any, replacement: Any, ctx: Context
+) -> VyList:
```
Comment (Steffan153) [inline vyxal/helpers.py:932]: what's this unused useless helper lmao

Comment (ysthakur) [reply vyxal/helpers.py:932]: Yeah I forgot to delete that

**Comment by ysthakur** [comment]
bruh I made a whole generic replace method that could recurse into lists and start at any occurrence and have a max number of occurrences. What you want is much easier though, I'll implement that

**Comment by Steffan153** [comment]
what you did could be useful too though, you cuold put it into a digraph

**Comment by ysthakur** [comment]
Okay, if it should be working now. I made it so `"abcd", ["a", "b"], ["foo"]` is equivalent to `"abcd", ["a", "b"], ["foo", ""]` if that's okay (the replacements list gets extended with empty strings).

File: documents/knowledge/elements.yaml [hunk 2]
```diff
@@ -5124,7 +5127,7 @@
   description: Replace the first instance of an item with another item
   arity: 3
   overloads:
-    any-any-any: a.replace_first(b, c)
+    any-any-any: a.replace(b, c, count=1). See "V" (Replace First) for specifics.
```
Comment (Steffan153) [inline documents/knowledge/elements.yaml:5130]:
```suggestion
    any-any-any: a.replace(b, c, count=1). See "V" (Replace) for specifics.
```

**Review by Steffan153 — COMMENTED** [review]
It doesn't look like this works if `lhs` is a list

**Comment by ysthakur** [comment]
@Steffan153 You mean like `[1, 2, 3, 4], [1, 2], [5, 6] -> [5, 6, 3, 4]`? I don't think it should work like that for lists, in case someone wants to replace something in a nested list like `[[1, 2], [0, 1], [3, 4]], [[1, 2], [3, 4]], [[1, 0], [2, 3]] -> [[1, 0], [0, 1], [2, 3]]`

**Review by Steffan153 — COMMENTED** [review]
What if somebody wants to replace a whole list in a list? I'd say put the recursive behavior into a digraph

**Comment by ysthakur** [comment]
> What if somebody wants to replace a whole list in a list?

What about it? This PR doesn't change the behavior for lists

> I'd say put the recursive behavior into a digraph

I'll get around to it after we merge this :)

**Review by Lyxal — APPROVED** [review]
This approval is for when the intended behaviour is reached. The overall infrastructure of the change lgtm

**Comment by Lyxal** [comment]
No merging yet

**Comment by Steffan153** [comment]
> What about it? This PR doesn't change the behavior for lists

Okay so why not implement this the same way you implementented strings?

**Comment by ysthakur** [comment]
@Steffan153 In case someone wants to replace a whole list inside a nested list. What if someone wants to replace an entire list in a nested list like `[[1, 2], [0, 1], [3, 4]], [1, 2], [1, 0] -> [[1, 0], [0, 1], [3, 4]]`? If it worked for lists the way it did for strings, it would instead do nothing. Do we want that behavior? I guess we could make a digraph for the current behavior on lists.

**Review by Steffan153 — APPROVED** [review]
I guess that makes sense.

**Pull request merged**